app/src/main/java/org/merakilearn/ui/onboarding/WelcomeFragment.kt

```diff
@@ -74,6 +80,63 @@ class WelcomeFragment : BaseFragment() {
         })
     }
 
+    private fun initGoogleSignInOption() {
+        val gso =
+            GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail()
+                .requestIdToken(getString(R.string.server_client_id))
+                .build()
+        mGoogleSignInClient = GoogleSignIn.getClient(requireActivity(), gso)
+    }
+
+    private fun signInWithGoogle() {
+        val signInIntent = mGoogleSignInClient!!.signInIntent
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == RC_SIGN_IN) {
+            val task =
+                GoogleSignIn.getSignedInAccountFromIntent(data)
+            handleSignInResult(task)
+        }
+    }
+
+    private fun handleSignInResult(completedTask: Task<GoogleSignInAccount>) {
+        try {
+            val account = completedTask.getResult(ApiException::class.java)
+            logOutSelectedAccount(account)
+        } catch (e: ApiException) {
+            Log.e(LoginFragment.TAG, "signInResult:failed code=", e)
+            toast(getString(R.string.unable_to_sign))
+        }
+    }
+
+    private fun logOutSelectedAccount(account: GoogleSignInAccount) {
+        mGoogleSignInClient?.signOut()?.addOnSuccessListener {
+            viewModel.idToken.value = account.idToken
```

Comment:
liveData should be used to communicate from viewModel to view not the other way around. This should be done by calling a method on viewModel or better use already implemented actions for the sake of consistency

Reply:
Since the live data observer activate on condition , so Transformation.switchMap is used. To do that, above code is written

Reply:
I guess we can remove switch map, it seems unnecessary. To use live data to pass some values to viewModel is not a good approach.

Reply:
How can we then do ? I saw same approach in the YouTube video, I shared. Please suggest me some other way then

Reply:
Is it okay if I make these changes and make a commit??